framework/src/base/Assembly.C

```diff
@@ -2233,21 +2234,20 @@ Assembly::init(const CouplingMatrix * cm)
   if (_block_diagonal_matrix && scalar_vars.size() != 0)
     _block_diagonal_matrix = false;
 
-  auto num_vector_tags = _subproblem.numVectorTags();
```

Comment:
`auto num_vector_tags = _residual_vector_tags.size();`, and we do not need to change that many lines of code